receiver/hostmetricsreceiver/internal/scraper/processscraper/process_scraper.go

```diff
@@ -239,3 +244,13 @@ func (s *scraper) scrapeAndAppendDiskIOMetric(now pcommon.Timestamp, handle proc
 
 	return nil
 }
+
+func (s *scraper) scrapeAndAppendThreadsMetrics(now pcommon.Timestamp, handle processHandle) error {
+	threads, err := handle.NumThreads()
```

Comment:
Given that it's metric disabled by default, let's check if user enabled it before touching the handle to avoid possibly unrelated log error logs

```
if !s.config.Metrics. ProcessThreads.Enabled {
	return nil
}
```